analyzers/src/SonarAnalyzer.Common/Common/AnalyzerConfiguration.cs

```diff
@@ -84,20 +85,20 @@ public void Initialize(AnalyzerOptions options)
 
             public void Initialize(AnalyzerOptions options)
             {
-                var currentSonarLintXmlPath = GetSonarLintXmlPath(options);
-                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXmlPath)
+                var currentSonarLintXmlFile = GetSonarLintXmlFile(options);
+                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXmlFile?.Path)
```

Comment:
you do the null-check three times - maybe store the path in a local var?

Reply:
discussed offline with Rita that anyhow this method deserves more refactoring and this PR is minimal to unblock SonarLint for Rider